editor.planx.uk/src/@planx/components/DateInput/model.ts

```diff
@@ -68,32 +69,44 @@ export const parseDate = (date?: string) => {
 /**
  * Validates that a given string is a date in the correct format
  */
-export const dateSchema = () => {
+export const dateSchema = ({ required }: { required: boolean }) => {
   return string()
     .test("missing day", "Date must include a day", (date?: string) => {
+      if (!required && !date) return true;
```

Comment:
I spent a while looking into this - there's not a hierarchical way to run this check once and first - it needs to be repeated. Pretty tedious / error prone!